### ISSUE TITLE
Testing OpenMPI Optimization flags on Milan

### ISSUE THREAD

**Issue by mathomp4** [GEOS-ESM/GEOSldas#691]
Recently @wmputman found some useful Open MPI flags to speed up performance of GEOSgcm on the Milans as seen in https://github.com/GEOS-ESM/GEOSgcm_App/pull/543:

```
# Turn off warning about TMPDIR on NFS
setenv OMPI_MCA_shmem_mmap_enable_nfs_warning 0
# pre-connect MPI procs on mpi_init
setenv OMPI_MCA_mpi_preconnect_all 1
setenv OMPI_MCA_coll_tuned_bcast_algorithm 7
setenv OMPI_MCA_coll_tuned_scatter_algorithm 2
setenv OMPI_MCA_coll_tuned_reduce_scatter_algorithm 3
setenv OMPI_MCA_coll_tuned_allreduce_algorithm 3
setenv OMPI_MCA_coll_tuned_allgather_algorithm 4
setenv OMPI_MCA_coll_tuned_allgatherv_algorithm 3
setenv OMPI_MCA_coll_tuned_gather_algorithm 1
setenv OMPI_MCA_coll_tuned_barrier_algorithm 0
# required for a tuned flag to be effective
setenv OMPI_MCA_coll_tuned_use_dynamic_rules 1
# disable file locks
setenv OMPI_MCA_sharedfp "^lockedfile,individual"
```

It might be interesting to see if GEOSldas-on-Milan benefits as well. From what I see of GEOSldas, I think they'd go into [`lenkf.j.template`](https://github.com/GEOS-ESM/GEOSldas/blob/develop/src/Applications/LDAS_App/lenkf.j.template) as I see in there an Intel MPI flag:

https://github.com/GEOS-ESM/GEOSldas/blob/61d59b73a6603a8fe517a47346b76a1d16237730/src/Applications/LDAS_App/lenkf.j.template#L43

(which, by the way, is an obsolete flag so it could be removed).

We'd probably want to do some "if openmpi, else if intelmpi" blocks in there in the final form, but as a test on the Milans, it could just be hardcoded. (That said, I might be looking at entirely the wrong file as well. GEOSldas still confuses me at times. 😄 )

Note that in my testing, the flags didn't seem to hurt GEOSgcm GNU on SLES12, so that's nice, it just made the Milan runs less slow.

**Comment by mathomp4** [comment]
One more update from https://github.com/GEOS-ESM/GEOSgcm_App/pull/543, it turns out the PSM3 provider for Intel MPI seems to be pretty good.

**Comment by gmao-rreichle** [comment]
addressed in #693
